### PR TITLE
refactor(localization): localize validation methods

### LINKED ISSUE #692: Missing i18n in kafka package
## Description

pkg/kafka has many validators that return non externalized strings.
Examples:

https://github.com/redhat-developer/app-services-cli/blob/srs-management-cli/pkg/kafka/kafkaerr/error.go#L30:L30

Creating a separate issue to address outside my current PR.

### PR DESCRIPTION
**Pull request by rkpattnaik780**
Localization has been added to validation methods as needed.

Closes #692

### Verification steps

The PR affects the following commands:

`rhoas kafka create`
`rhoas kafka topic create`
`rhoas kafka topic list --search <search input>`
`rhoas kafka topic update`
`rhoas kafka serviceaccount create`

### Type of change

- [] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [X] Refactor

### Checklist

- [ ] Documentation added for the feature
- [X] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: pkg/kafka/topic/validators_test.go [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+
+	"github.com/redhat-developer/app-services-cli/pkg/localize"
+	"github.com/redhat-developer/app-services-cli/pkg/localize/goi18n"
+)
+
+var validator *Validator = &Validator{
+	Localizer: getValidator(),
+}
+
+func getValidator() localize.Localizer {
```
Comment (rkpattnaik780) [inline pkg/kafka/topic/validators_test.go:14]: Is there a better way to initialize the localizer in test files.

File: pkg/cmd/kafka/create/create.go [hunk 2]
```diff
@@ -53,6 +52,7 @@ type Options struct {
 	Connection factory.ConnectionFunc
 	Logger     func() (logging.Logger, error)
 	localizer  localize.Localizer
+	validator  *pkgKafka.Validator
```
Comment (craicoverflow) [inline pkg/cmd/kafka/create/create.go:55]: Hmm, i would say there is no need to add the localizer to `opts` - just initialize it when you need it.

File: pkg/kafka/kafka_util.go [hunk 3]
```diff
@@ -18,16 +20,21 @@ var (
...
 )
 
+type Validator struct {
+	Localizer  localize.Localizer
+	Connection factory.ConnectionFunc
+}
+
 // ValidateName validates the proposed name of a Kafka instance
-func ValidateName(val interface{}) error {
+func (v Validator) ValidateName(val interface{}) error {
```
Comment (craicoverflow) [inline pkg/kafka/kafka_util.go:29]:
```suggestion
func (v *Validator) ValidateName(val interface{}) error {
```

it should use a pointer preferably.

File: pkg/kafka/kafka_util.go [hunk 4]
```diff
@@ -86,20 +93,25 @@ func ValidateSearchInput(val interface{}) error {
 }
 
 // ValidateNameIsAvailable checks if a kafka instance with the given name already exists
-func ValidateNameIsAvailable(api kafkamgmtclient.DefaultApi, localizer localize.Localizer) func(v interface{}) error {
-	return func(v interface{}) error {
-		name, _ := v.(string)
+func (v Validator) ValidateNameIsAvailable(val interface{}) error {
```
Comment (craicoverflow) [inline pkg/kafka/kafka_util.go:96]:
```suggestion
func (v *Validator) ValidateNameIsAvailable(val interface{}) error {
```

File: pkg/kafka/topic/validators_test.go (hunk 1 above)
Comment (craicoverflow) [reply pkg/kafka/topic/validators_test.go:14]: You can use a `func init() {}` function which should be run before everything else in a Go file.

File: pkg/kafka/kafka_util.go [hunk 5]
```diff
@@ -18,16 +20,21 @@ var (
 	validSearchRegexp = regexp.MustCompile(`^([a-zA-Z0-9-_%]*[a-zA-Z0-9-_%])?$`)
 )
 
+type Validator struct {
```
Comment (craicoverflow) [inline pkg/kafka/kafka_util.go:23]: Public types and functions should always be documented.

File: pkg/kafka/kafka_util.go (hunk 4 above)
Comment (craicoverflow) [reply pkg/kafka/kafka_util.go:96]: Neat :D

**Review by craicoverflow — CHANGES_REQUESTED** [review]
Getting an error:

```shell
❯ ./rhoas kafka topic create topic-1
Error: could not cast &{0xc0003a1860}, to type "int32"
```

**Comment by craicoverflow** [comment]
> Bug fix (non-breaking change which fixes an issue)

Just so you know this is not a bug fix!

**Comment by craicoverflow** [comment]
Could you add verification steps so I know which commands I need to check, otherwise I need to inspect each file to work it out.

File: pkg/cmd/kafka/topic/create/create.go [hunk 6]
```diff
@@ -229,8 +238,8 @@ func runInteractivePrompt(opts *Options) (err error) {
 		promptName,
 		&opts.topicName,
 		survey.WithValidator(survey.Required),
-		survey.WithValidator(topicutil.ValidateName),
-		survey.WithValidator(topicutil.ValidateNameIsAvailable(api, kafkaInstance.GetName())),
+		survey.WithValidator(validator.ValidateName),
+		survey.WithValidator(topicutil.ValidateNameIsAvailable(api, kafkaInstance.GetName(), opts.localizer)),
```
[the remark this answers is not in the document]
Comment (rkpattnaik780) [reply pkg/cmd/kafka/topic/create/create.go:242]: Should we create another key in struct to pass the instance name?

[the remark this answers is not in the document]
Comment (craicoverflow) [reply pkg/cmd/kafka/topic/create/create.go:242]: The implementation is up to you. I probably would do that yes.

File: pkg/cmd/kafka/create/create.go [hunk 7]
```diff
@@ -24,7 +23,6 @@ import (
 	"github.com/AlecAivazis/survey/v2"
 	"github.com/redhat-developer/app-services-cli/pkg/dump"
 	"github.com/redhat-developer/app-services-cli/pkg/iostreams"
-
 	pkgKafka "github.com/redhat-developer/app-services-cli/pkg/kafka"
```
[the remark this answers is not in the document]
Comment (rkpattnaik780) [reply pkg/cmd/kafka/create/create.go:26]: Will di it in a follow up.

**Review by craicoverflow — APPROVED** [review]

**Pull request merged**